docs/src/pages/[platform]/connected-components/account-settings/change-password/react.mdx

```diff
@@ -1,3 +1,129 @@
+  
+  return (
+    <ChangePassword onSuccess={handleSuccess}/>
+  );
+}
+```
+
+### Custom Validation
+
+You can override the default custom password validator with your own. To do so, you can use the validators prop, which takes an array of ValidatorOptions, which is of type:
+
+```ts
+interface ValidatorOptions {
+  validationMode: 'onBlur' | 'onChange' | 'onTouched';
+  validator: (value: string) => boolean;
+  message: string;
+}
+```
+
+- `onBlur` validates password on every blur event
+- `onChange` validates password on every change event
+- `onTouched` validates password on first blur, and on every change event after the first blur.
+
+For example, following is a minLength validator that validates on every change:
+
+```js
+const minLength = {
+  validationMode: 'onChange',
+  validator: (password) => password.length >= 4,
+  message: 'Password must have length 4 or greater',
+};
+```
+
+You can pass multiple validators to ConfirmPassword to override the default validator:
+
+```jsx
+import React from 'react';
+import { ChangePassword } from '@aws-amplify/ui-react';
+
+function App() {
+  const minLength = {
+    validationMode: 'onChange',
+    validator: (password) => password.length >= 4,
+    message: 'Password must have length 4 or greater',
+  };
+  
+  const handleSuccess = () => {
+    alert('password is successfully changed!')
+  }
+  
+  return (
+    <ChangePassword onSuccess={handleSuccess} validators={[minLength]}/>
+  );
+}
+```
+
+### Component Overrides
+
+You can provide your own UI components to override parts of `ChangePassword`. It supports the following slots:
+
+<ReactPropsTable props={OVERRIDES} />
+
+#### Reusing default components
+
+Default components are accessible through `ChangePassword.COMPONENT_NAME` (e.g. `ChangePassword.NewPassword`) for your convenience. This is useful if you're interested in modifying just a small part of UI instead of overriding the whole component.
+
+```jsx
+function App() {
+  const components = {
+    // reuse default component
+    NewPassword: (props) => (
+      <ChangePassword.NewPassword 
+        {...props} 
+        label="Custom Label" 
+      />
+    ),
+  }
+
+  return (
+    <ChangePassword components={components} />
+  );
+}
+```
+
+#### Using your own component
+
+You can also provide your own custom to override components. 
+
+```jsx
+function App() {
+  const components = {
+    // use your own component
+    ConfirmPassword: ({ validationErrors, ...rest }) => (
+      <>
+        <input {...rest} type="password" />
+        {validationErrors.map((error) => (
+          <p key={error} role="alert">{error}</p>
+        ))}
+      </>
+    )
+  };
+  
+  return (
+    <ChangePassword components={components} />
+  );
+}
+```
+
+Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name`` are attached correctly. If you're providing your own custom components, make sure props are spread onto your input or button component.
```

Comment:
```suggestion
Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name` are attached correctly. If you're providing your own custom components, make sure props are spread onto your input or button component.
```